### ISSUE TITLE
future: display more information / source with NoCandidateFound error

### ISSUE THREAD

**Issue by blueyed** [jazzband/pip-tools#165]
I got the following error, but it was not obvious where it was coming from (which package has this dependency):

```
Could not find a version that matches factory-boy>=2.6.0
Tried: 1.0.0, 1.0.4, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.1.4, 1.1.5, 1.2.0, 1.3.0, 2.0.0, 2.0.1, 2.0.2, 2.1.0, 2.1.1, 2.1.2, 2.2.0, 2.2.1, 2.3.0, 2.3.1, 2.4.0, 2.4.1, 2.5.0, 2.5.1, 2.5.1, 2.5.2, 2.5.2
```

`pip-compile` should provide more information in that case.

I've taken a quick look via `ipdb`, but could not figure out quickly where to get this information from the `ireq` object.

(it was pulled in via `-e git+https://github.com/rbarrois/django-factory_boy.git#egg=django_factory_boy`)

**Comment by blueyed** [comment]
With the case at hand, it would be nice if pip-compile would handle this better.

pip-tools could maybe fallback to using the most recent version and issue a warning?!
(Maybe a command line switch could be required for this?)

Currently it aborts and fails to compile the requirements altogether.

The issue is that I am installining django-factory_boy from Git (because of required fixes), and there it requires the upcoming/unreleased version of another package (factory_boy).  This is (hopefully) rather uncommon (and has been reported in https://github.com/rbarrois/django-factory_boy/issues/3), but pip-compile could be more helpful in handling it.

**Comment by blueyed** [comment]
This would be especially useful in case there are multiple packages involved.

I was just seeing this:

> Could not find a version that matches djangorestframework<=3.1.2,==3.1.3,>=2.3.8,>=3.0.4

It would be helpful to see directly where the `<=3.1.2` is coming from.

**Comment by atugushev** [comment]
Closed in #837.
